src/components/Card.js

```diff
@@ -1,22 +1,47 @@
 export default class Card {
-  constructor({ data, handleImageClick }, cardSelector) {
+  constructor({ data, myId, handleLikeClick, handleDeleteClick, handleImageClick}, cardSelector) {
     this._data = data;
     this._name = data.name;
     this._link = data.link;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._id = data._id;
+    this._myId = myId;
+    this._handleLikeClick = handleLikeClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._likes = data.likes;
+    this._ownerId = data.owner._id;
   }
 
+  getId() {
+    return this._id;
+  }
+  isLiked() {
+    return this._likes.some((like) => like._id === this._myId);
+  }
+
+  setLikes(likes) {
+    this._likes = likes;
+    this._handleLikeIcon();
+    this.renderLikes();
+  }
+  renderLikes() {
+    this._likesAmount = this._cardElement.querySelector(".card__like-amount");
+    this._likesAmount.textContent = this._likes.length;
+  };
```

Comment:
You should handle like icon inside this function, so you wan't need to call two functions when updating likes view

Reply:
Done